nextest-runner/src/target_runner.rs

```diff
@@ -168,11 +168,25 @@ impl TargetRunner {
                     configs.push(home_config);
                 }
             }
+        } else {
+            // Remove cargo home if the current directory is underneath
+            // the same root as $CARGO_HOME
+            let mut cargo_home_path = home::cargo_home_with_cwd(root.as_std_path())
+                .ok()
+                .and_then(|home| Utf8PathBuf::from_path_buf(home).ok());
+            if let Some(mut cargo_home_path) = cargo_home_path.as_mut() {
+                if let Some(home_config) = read_config_dir(&mut cargo_home_path) {
+                    if let Some(p) = configs.iter().position(|path| path == &home_config) {
+                        configs.remove(p);
+                    }
+                }
+            }
```

Comment:
Honestly looking at this, I think the correct semantics for tests are something like "isolate it to the project root and don't iterate further up".

Reply:
I don't really know how to do this, I don't think there is an easy way of finding the workspace root (from cargo env-vars at least).

Reply:
We do have it in the `PackageGraph`.

Reply:
Hmm, we do in https://github.com/nextest-rs/nextest/blob/main/nextest-runner/tests/integration/fixtures.rs#L89-L106 don't we?

Reply:
I'd just pass in something like `terminate_search_at: Option<&Utf8Path>`.

Reply:
Oh I see, so there's two separate things here:
* Terminate search at the workspace root.
* Don't read cargo_home.

Given that the use cases for these two are linked to each other (isolation for tests) I think it makes sense to combine them into one.

Reply:
done